### PR TITLE
test: logging: improve test coverage for logging subsystem

### PR DESCRIPTION
**Pull request by mengxianglinx**
Add some test point in existing test cases. Add some test
cases for usersapce.

Signed-off-by: Meng xianglin <xianglinx.meng@intel.com>

### CODE REVIEW

**Review by LeiW000 — APPROVED** [review]

File: tests/subsys/logging/log_core_additional/testcase.yaml [hunk 1]
```diff
@@ -5,3 +5,8 @@ tests:
   logging.add.sync:
     tags: logging
     extra_args: CONF_FILE=log_sync.conf
+  logging.add.user:
+    tags: logging
+    platform_exclude: mps2_an521 mps2_an385
```
Comment (enjiamai) [inline tests/subsys/logging/log_core_additional/testcase.yaml:10]: Is there any reason to exclude them?

**Review by enjiamai — COMMENTED** [review]
Look good, only one question, pls see the comment above.

File: tests/subsys/logging/log_core_additional/testcase.yaml (hunk 1 above)
Comment (mengxianglinx) [reply tests/subsys/logging/log_core_additional/testcase.yaml]: updated. build error occurs, I've reported an issue

**Review by enjiamai — APPROVED** [review]
@mengxianglinx  Anas has already solve the  it8xxx2_evb build fail issue you submitted,  https://github.com/zephyrproject-rtos/zephyr/pull/31506, please rebase it~

**Comment by ioannisg** [comment]
@nordic-krch please review this change.

**Review by nordic-krch — APPROVED** [review]

**Pull request merged**